packages/core/src/analytics/analytics-step-tracker.ts

```diff
@@ -421,15 +421,18 @@ export default class AnalyticsStepTracker implements StepTracker {
             }
 
             return {
-                product_id: item.productId,
+                product_id: String(item.productId),
                 sku: item.sku,
                 price: item.salePrice,
                 image_url: item.imageUrl,
                 name: item.name,
                 quantity: item.quantity,
-                brand: itemsData[item.productId] ? itemsData[item.productId].brand : '',
                 category: itemsData[item.productId] ? itemsData[item.productId].category : '',
-                variant: (itemAttributes || []).join(', '),
+                variant: (itemAttributes || []).join(', ') || 'single-product-option',
```

Comment:
Can you please refactor this to move to a separate function, and also break it into multiple lines since it can be a bit hard to read.

Reply:
Hey @animesh1987,
Refactored this part a bit and also moved all this transform logic to a separate function, so it should be more straightforward now. Please, check again.